### PR TITLE
Upon beating boss party, return boss names rather than boss party name.

### PR DESCRIPTION
**Pull request by Anexes**
Rather than their actual names, bosses still seem to use their party name (if they have one,) in combat logs. Therefore, they still return with Character.BossKills.Wardens (for example,) but it does go through and increment once for each boss. Now all that's left is to fix it so they use their actual names rather than their party names.

### CODE REVIEW

File: src/plugins/events/events/BattleBoss.js [hunk 1]
```diff
@@ -54,7 +54,10 @@ export class BattleBoss extends Event {
     if(!battle.isLoser(player.party) && !battle._isTie) {
       _.each(player.party.players, p => {
         if(!p.$statistics) return;
-        p.$statistics.incrementStat(`Character.BossKills.${bossName}`);
+
```
Comment (seiyria) [inline src/plugins/events/events/BattleBoss.js]: please fix the indentation

File: src/plugins/events/events/BattleBoss.js [hunk 2]
```diff
@@ -54,10 +54,10 @@ export class BattleBoss extends Event {
     if(!battle.isLoser(player.party) && !battle._isTie) {
       _.each(player.party.players, p => {
         if(!p.$statistics) return;
-		
-		_.each(_.keys(bosses), (boss) => {
-			p.$statistics.incrementStat(`Character.BossKills.${boss}`);
-		});
+
```
Comment (seiyria) [inline src/plugins/events/events/BattleBoss.js]: still off by one indent level

**Comment by Anexes** [comment]
@seiyria I wasn't able to work out how to test it myself; got an error for not having a .env file but I've been super frustrated about code stuff all day so I decided to put that off 'til some other time. Then Joe mentioned the Wardens thing and I got sucked back in trying to fix it.

It currently sets boss.stats.name to the value pulled from the tile, which is the boss party name. As far as I can tell, member should be a string (since it's used as a key for `Bosses[member]`) so I *think* `boss.stats.name = member` should work. But then, I never know with JS.

Long story short, I think it should work, but I'm sorry if it doesn't.

**Comment by Anexes** [comment]
Did the lint thing and ran with 100 players for 30 minutes. Seems like 100 was too many, 'cause everything slowed down when I tried to terminate it and it tossed some errors but it otherwise had no problems that I could see. You said the errors were just from terminating the process, but I didn't get them when I tested my other PR, if that means anything.

**Comment by Anexes** [comment]
I'm testing it again now with 75 players, which seemed to work with my other PR, to see if I get the same errors at the end. If it all goes well, I'll ping for a merge.

**Comment by seiyria** [comment]
Yep. You'll also want to check the statistics to make sure they're being recorded correctly. I'm not sure how many boss parties there are but you'll want to warp all your people near one and then check the statistics afterwards. Something like:

```
db.players.update({}, { $set: { x: 10, y: 10, map: 'Norkos -10' } }, { multi: true });
```

You can do that from the mongo console (open your terminal and run `mongo`, then paste that in). For this one, just having it run isn't enough unfortunately because if it's not recording statistics correctly it doesn't matter.

Also, apparently I started a review instead of submitting a comment, so you'll want to review that and make that change.

File: src/plugins/events/events/BattleBoss.js [hunk 3]
```diff
@@ -55,9 +55,9 @@ export class BattleBoss extends Event {
       _.each(player.party.players, p => {
         if(!p.$statistics) return;
     
-      _.each(_.keys(bosses), (boss) => {
-        p.$statistics.incrementStat(`Character.BossKills.${boss}`);
-      });
+        _.each(_.keys(bosses), (boss) => {
+          p.$statistics.incrementStat(`Character.BossKills.${boss}`);
```
Comment (seiyria) [inline src/plugins/events/events/BattleBoss.js]: Use whichever is accurate based on your changes, but keep it in mind either way because `boss` is an object.

Comment (Anexes) [inline src/plugins/events/events/BattleBoss.js]: @seiyria I didn't change it from `boss.stats.name`, I just changed the value it's assigned. I've tried searching for `boss.name` but GitHub doesn't recognize periods in a search, so I can't tell if it's used like that elsewhere. I'm pretty sure it should be `boss.stats.name`.

However, I just noticed something; is there anything preventing bosses from getting [unique names](https://github.com/IdleLands/IdleLands/blob/master/src/shared/monster-generator.js#L124-L130)? That would interfere with the whole statistics thing.

Comment (seiyria) [inline src/plugins/events/events/BattleBoss.js]: No, nothing is stopping them. You'd have to record the original name when assigning it, then use that instead.

Comment (seiyria) [inline src/plugins/events/events/BattleBoss.js]: Yes, they can get unique names. You should store the base name on the boss as a property like... `boss._name` and then use that when recording stats.

**Comment by Anexes** [comment]
@seiyria Typing `mongo` into the console doesn't do anything. What am I missing?

**Comment by seiyria** [comment]
Oh right. Uh, you would have to go into the mongo directory and open a CMD there and do it, which should work, since mongo wasn't added to your PATH.

**Comment by Anexes** [comment]
Only took me like 3 hours of trying to figure out how to use the mongo shell to get a readable statistics list and 2 hours of trying to figure out why I was getting "BossKills": [ 1 ], but it finally works!Got some characters to kill Warden of Norkos (x2) as well as the elves from Santa's workshop. Individual bosses work fine as well.

I've got the final statistics log here if you want to look at it: [Statistics.txt](https://github.com/IdleLands/IdleLands/files/526567/Statistics.txt)

**Comment by Anexes** [comment]
@seiyria Uh, sorry if this pings you twice. Wanted to make sure it got to you but I forgot to put the @ in until I had already hit comment and I'm not sure if a retroactive ping actually...pings.

**Comment by seiyria** [comment]
thanks for testing it!
